skill-tree/src/main/java/com/RDS/skilltree/services/external/RdsServiceImplementation.java

```diff
@@ -0,0 +1,22 @@
+package com.RDS.skilltree.services.external;
+
+import com.RDS.skilltree.viewmodels.RdsUserViewModel;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+@RequiredArgsConstructor
+public class RdsServiceImplementation implements RdsService {
+    private final RestTemplate restTemplate;
+
+    @Value("${rds.backendBaseUrl}")
+    private String rdsBackendBaseUrl;
+
+    @Override
+    public RdsUserViewModel getUserDetails(String id) {
```

Comment:
Here url is being constructed by string concatenation. This approach can be vulnerable to security issues like https://en.wikipedia.org/wiki/SQL_injection if id is not properly sanitized.
Consider using a parameterized query with the restTemplate.